src/javascript/esutil.ts

```diff
@@ -428,3 +428,153 @@ function* _getMethods(node: babel.Node) {
     }
   }
 }
+
+export function extractPropertyFromGetterOrSetter(
+    method: babel.ClassMethod|babel.ObjectMethod,
+    jsdocAnn: jsdoc.Annotation|undefined,
+    document: JavaScriptDocument): ScannedProperty|null {
+  if (babel.isClassMethod(method) && method.static) {
+    return null;
+  }
+
+  if (method.key.type !== 'Identifier') {
+    return null;
+  }
+
+  if (method.key.name === undefined) {
+    return null;
+  }
+
+  let type;
+  let description;
+  let privacy: Privacy = 'public';
+  let readOnly = false;
+
+  if (jsdocAnn) {
+    const ret = getReturnFromAnnotation(jsdocAnn);
+    type = ret ? ret.type : undefined;
+    description = jsdoc.getDescription(jsdocAnn);
+    privacy = getOrInferPrivacy(method.key.name, jsdocAnn);
+    readOnly = jsdoc.hasTag(jsdocAnn, 'readonly');
+  }
+
+  return {
+    name: method.key.name,
+    astNode: method,
+    type,
+    jsdoc: jsdocAnn,
+    sourceRange: document.sourceRangeForNode(method)!,
+    description,
+    privacy,
+    warnings: [],
+    readOnly,
+  };
+}
+
+export function extractPropertiesFromNode(
```

Comment:
How about `extractPropertiesFromAccessorsAndObjectLiteralBodies`?

I know it's an absurdly long name, but we have _so many_ ways of determining properties, it's a challenge to tell when reading a method what any given call is trying to do.

Reply:
> I know it's an absurdly long name

You don't say 😂 

i've changed it to `extractPropertiesFromClassOrObjectBody` (is that fine?)

Reply:
+1